src/main/java/spoon/support/reflect/reference/CtTypeParameterReferenceImpl.java

```diff
@@ -50,6 +50,7 @@
 
 	public CtTypeParameterReferenceImpl() {
 		super();
+		this.setBoundingType(getFactory().Type().objectType()); // by default set bounding type to object
```

Comment:
we should create new CtTypeReference, because AST should not share nodes

Reply:
Actually the methods already create a clone of OBJECT in `TypeFactory`, see: https://github.com/INRIA/spoon/blob/e2e359e2e59c23a94cd042a6851e8058cd62dc5c/src/main/java/spoon/reflect/factory/TypeFactory.java#L250

So I assume the nodes are not shared.

Reply:
you are right. I did not knew that. So it is OK